app/src/main/java/xyz/jhughes/laundry/adapters/MachineAdapter.java

```diff
@@ -135,24 +145,134 @@ public void onClick(DialogInterface dialog, int which) {
+        //Constructs the dialog to wait for a machine
+        //checks the server while the dialog is open and the app is running in the background
+        final Machine m2 = m;
+        final Handler handler = new Handler();
+        lockListener.onLock();
+        AlertDialog.Builder machineWaitingDialog = new AlertDialog.Builder(mContext);
+        machineWaitingDialog.setTitle(mContext.getString(R.string.alarm))
+                .setMessage(mContext.getString(R.string.available_timer_message1) + " " + m.getName() + " " + mContext.getString(R.string.available_timer_message2))
+                .setCancelable(true)
+                .setPositiveButton(mContext.getString(R.string.available_timer_refresh), null)
+                .setNegativeButton(mContext.getString(R.string.available_timer_cancel), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+                        handler.removeCallbacksAndMessages(null);
+                        dialog.cancel();
+                    }
+                });
+        //add message to load machine before
+        LayoutInflater inflater = LayoutInflater.from(mContext);
+        View q = inflater.inflate(R.layout.view_available_machine, null);
+        machineWaitingDialog.setView(q);
+        TextView number = (TextView) q.findViewById(R.id.machine_name_number);
+        number.setText(getNumberFromName(m));
+        final AlertDialog alertDialog = machineWaitingDialog.create();
+        alertDialog.show();
+        alertDialog.setOnCancelListener(new DialogInterface.OnCancelListener() {
+            @Override
+            public void onCancel(DialogInterface dialog) {
+                lockListener.onUnlock();
+            }
+        });
+        alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String apiLocationFormat = Constants.getApiLocation(MachineAdapter.this.roomName);
+                Call<ArrayList<Machine>> call = MachineService.getService().getMachineStatus(apiLocationFormat);
+                call.enqueue(new Callback<ArrayList<Machine>>() {
+                    @Override
+                    public void onResponse(Call<ArrayList<Machine>> call, Response<ArrayList<Machine>> response) {
+                        ArrayList<Machine> body = response.body();
+                        if (body.contains(m2)){
+                            Machine m3 = body.get(body.indexOf(m2));
+                            if (m3.getStatus().equals(MachineStates.IN_USE)){ //
+                                createNotification(m3);
+                                alertDialog.cancel();
+                            }
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<ArrayList<Machine>> call, Throwable t) {
```

Comment:
Added failure call to here and MachineCheckerRunnable.  Is there a category you want to use to see when people hit refresh or cancel out of the dialog, or when the timer starts automatically?